### PR TITLE
Added Graph Neural Network Classification

### PR DESCRIPTION
**Pull request by LindaSt**
**New runner for Graph Classification:**
- Dataset has to be in gxl format (see IAM graph database), with the train, test and validation set split as well as the class labels in a cxl file each
- Uses additional packages from https://github.com/rusty1s/pytorch_geometric -> see exten_environment.sh
- Specific data loader implemented
- New command-line arguments introduced specifically for GNNs

**New models for Graph Classification:**
- added several neural network architectures for graph classification

**Updated ReadME with the correct installation procedure**

### CODE REVIEW

File: template/runner/graph_classification_iamdb/graph_classification_iamdb.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+# Gale
+from template.runner.base import BaseRunner
+
+# Delegated
+from .setup import GraphClassificationSetup
+from .train import GraphClassificationTrain
+from .evaluate import GraphClassificationEvaluate
+
+
+class GraphClassificationIamdb(BaseRunner):
```
Comment (Renthal) [inline template/runner/graph_classification_iamdb/graph_classification_iamdb.py:13]: is this runner peculiar to the iamdb ? Can it be a general GNN runner?

File: template/runner/graph_classification_iamdb/graph_classification_iamdb_CL_arguments.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+# Torch
+import torch
+
+# Gale
+import models
+from template.runner.base.base_CL_arguments import BaseCLArguments
+
+
+class GraphClassificationIamdbCLArguments(BaseCLArguments):
```
Comment (Renthal) [inline template/runner/graph_classification_iamdb/graph_classification_iamdb_CL_arguments.py:12]: In image classification this file is called just: `CL_arguments` maybe we could rename it to be consistent

File: template/runner/graph_classification_iamdb/graph_classification_iamdb_inference.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# DeepDIVA
+from template.runner.base import BaseInference
+from template.runner.image_classification.setup import ImageClassificationSetup
+
+
+class GraphClassificationInference(BaseInference):
+
+    def __init__(self):
+        super().__init__()
+        self.setup = ImageClassificationSetup()
```
Comment (Renthal) [inline template/runner/graph_classification_iamdb/graph_classification_iamdb_inference.py:10]: this should not be done, as if something is changed in ImageClassification it breaks here and this violates the runner incapsulation design

Comment (LindaSt) [reply template/runner/graph_classification_iamdb/graph_classification_iamdb_inference.py:10]: I have not actually tested/implemented this. I will delete this file for now and implement and test later.

Comment (Renthal) [reply template/runner/graph_classification_iamdb/graph_classification_iamdb_inference.py:10]: Good solution. Then you can delete the file and commit(+push) again

File: template/runner/graph_classification_iamdb/graph_classification_iamdb.py (hunk 1 above)
Comment (LindaSt) [reply template/runner/graph_classification_iamdb/graph_classification_iamdb.py:13]: It expects the same data structure as there is in the IAMDB database, with gxl files with the graphs and cxl files for the dataset splits and class labels. But I can definitely rename it to GraphClassificationGxl. Does that make more sense?

Comment (Renthal) [reply template/runner/graph_classification_iamdb/graph_classification_iamdb.py:13]: If it is a common standard it could make sense but i have no experience in GNN datasets so you will have to make the call

File: template/runner/graph_classification_iamdb/graph_classification_iamdb_CL_arguments.py (hunk 2 above)
Comment (LindaSt) [reply template/runner/graph_classification_iamdb/graph_classification_iamdb_CL_arguments.py:12]: Will be renamed in the second push.
